### PR TITLE
Improve the file upload example via SDK in docs

### LINKED ISSUE #20497: Improve documentation for File uploads via SDK
### Page

https://docs.directus.io/reference/files.html#upload-a-file

### Describe the Inaccuracy

The documentation for uploading files via SDK is missing how to specify the original filename.

```js
// Would expect
formData.append('file', raw_file, 'example.txt');
// instead of
formData.append('file', raw_file);
```

Also it would be nice to include an example on how to create the raw file object from a local file, e.g.
```js
const file = new Blob([readFileSync('/path/to/example.txt')], { type: 'text/plain' });
```

### PR DESCRIPTION
**Pull request by paescuj**
Fixes #20497

### CODE REVIEW

**Review by estheragbaje — APPROVED** [review]

**Review by br41nslug — CHANGES_REQUESTED** [review]
Am i missing something obvious about why it fails to build now? 🤔

File: docs/reference/files.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ readTime: 10 min read
 pageClass: page-reference
 ---
 
+<script setup lang="ts">
+import { data as packages } from '@/data/packages.data.js';
+</script>
```
[the remark this answers is not in the document]
Comment (paescuj) [reply docs/reference/files.md:9]: This fetches the information about packages in the monorepo, which should be used further down to display the current version of the SDK.
The issue however seems to come from the aliased import (`@/`) which apparently doesn't work on Windows (how, dear Windows?!). I'll remove the alias.

[the remark this answers is not in the document]
Comment (br41nslug) [reply docs/reference/files.md:9]: Adding a definition of what is supposed to be loaded to the `tsconfig.json` does resolve the issue of `pnpm dev` (but the build is still failing on the blog data)
```json
	"compilerOptions": {
		"baseUrl": ".",
		"paths": {
			"@/components/*": ["./.vitepress/components/*"],
			"@/data/*": ["./.vitepress/data/*"]
		}
	},
```

File: docs/reference/files.md [hunk 2]
```diff
@@ -597,22 +601,68 @@ Not supported by GraphQL
...
 import { createDirectus, rest, uploadFiles } from '@directus/sdk';
-import { readFileSync } from 'node:fs';
+import { readFile } from 'node:fs/promises';
 
 const client = createDirectus('https://directus.example.com').with(rest());
 
-const title = 'example';
-const file = new Blob([readFileSync('example.txt')]);
+const title = 'Example';
+const file = new Blob([await readFile('example.txt')], { type: 'text/plain' });
```
[the remark this answers is not in the document]
Comment (paescuj) [reply docs/reference/files.md:654]: It does 🤔 
https://github.com/directus/directus/blob/abf0cba75d650d7ccd91633ff2b6432b7afd6192/sdk/package.json#L56-L58

[the remark this answers is not in the document]
Comment (br41nslug) [reply docs/reference/files.md:654]: Lol thats not supposed to be in there i think (must have been a copy-paste 😇) this is however not enforced when simply using the sdk in my tests

image

File: docs/reference/files.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (paescuj) [reply docs/reference/files.md:9]: Fixed via #20545 👍 Thanks!

[the remark this answers is not in the document]
Comment (br41nslug) [reply docs/reference/files.md:9]: #20545 was merged before i was done reviewing 😬 found issue is in that PR

File: docs/reference/files.md [hunk 3]
```diff
@@ -597,22 +601,68 @@ Not supported by GraphQL
...
+::: code-group
+
+```js-vue [index.js]
+import { createDirectus, rest, uploadFiles } from 'https://unpkg.com/@directus/sdk@{{ packages['@directus/sdk'].version.major }}';
+
+const client = createDirectus('https://directus.example.com').with(rest());
+
+const form = document.querySelector('#upload-file');
+
+if (form && form instanceof HTMLFormElement) {
```
Comment (br41nslug) [inline docs/reference/files.md:615]:
```suggestion
const form = document.getElementById('upload-file');

if (form) {
```
Considering you have the HTML defined right there i think the HTMLFormElement check may be a little overkill.

Comment (paescuj) [reply docs/reference/files.md]: Copied that from an example for the older SDK, but I agree that this is a tad overkill 😄👍

**Review by br41nslug — APPROVED** [review]
Nothing more to nitpick from me 😇

**Pull request merged**